controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_types.go

```diff
@@ -53,5 +54,45 @@ func init() {
 
 // KubeadmControlPlaneTemplateResource describes the data needed to create a KubeadmControlPlane from a template.
 type KubeadmControlPlaneTemplateResource struct {
-	Spec KubeadmControlPlaneSpec `json:"spec"`
+	Spec KubeadmControlPlaneTemplateResourceSpec `json:"spec"`
+}
+
+// KubeadmControlPlaneTemplateResourceSpec defines the desired state of KubeadmControlPlane.
+// NOTE: This is a copy of the KubeadmControlPlaneSpec because not all fields of KubeadmControlPlane
+// make sense on the KubeadmControlPlaneTemplate (because some of them are managed by the topology
+// reconciler).
+type KubeadmControlPlaneTemplateResourceSpec struct {
+	// MachineTemplate contains information about how machines
+	// should be shaped when creating or updating a control plane.
+	MachineTemplate KubeadmControlPlaneTemplateMachineTemplate `json:"machineTemplate"`
+
+	// KubeadmConfigSpec is a KubeadmConfigSpec
+	// to use for initializing and joining machines to the control plane.
+	KubeadmConfigSpec cabpkv1.KubeadmConfigSpec `json:"kubeadmConfigSpec"`
+
+	// RolloutAfter is a field to indicate a rollout should be performed
+	// after the specified time even if no changes have been made to the
+	// KubeadmControlPlane.
+	//
+	// +optional
+	RolloutAfter *metav1.Time `json:"rolloutAfter,omitempty"`
+
+	// The RolloutStrategy to use to replace control plane machines with
+	// new ones.
+	// +optional
+	// +kubebuilder:default={type: "RollingUpdate", rollingUpdate: {maxSurge: 1}}
+	RolloutStrategy *RolloutStrategy `json:"rolloutStrategy,omitempty"`
+}
+
+// KubeadmControlPlaneTemplateMachineTemplate defines the template for Machines
+// in a KubeadmControlPlane object.
+// NOTE: This is a copy of the KubeadmControlPlaneMachineTemplate because not all fields of KubeadmControlPlaneMachineTemplate
```

Comment:
nit: "This is a copy of the KubeadmControlPlaneMachineTemplate" is misleading since it's not exactly a copy of it (I see how it makes sense from your perspective since you created it from `KubeadmControlPlaneMachineTemplate` but it could be confusing to users since a copy implies that they are identical which they are not)

Reply:
Good point. I rephrased both comments. PTAL :)